src/emulator/storage/rules/runtime.ts

```diff
@@ -318,7 +338,27 @@ export class StorageRulesRuntime {
         variables: runtimeVariables,
       },
     };
-    const response = (await this._sendRequest(runtimeActionRequest)) as RuntimeActionVerifyResponse;
+
+    return this._completeVerifyWithRuleset(opts.projectId, runtimeActionRequest);
+  }
+
+  private async _completeVerifyWithRuleset(
+    projectId: string,
+    runtimeActionRequest: RuntimeActionBundle,
+    overrideId?: number
+  ): Promise<{
+    permitted?: boolean;
+    issues: StorageRulesIssues;
+  }> {
+    const response = (await this._sendRequest(
+      runtimeActionRequest,
+      overrideId
+    )) as RuntimeActionVerifyResponse;
+
+    if ("context" in response) {
```

Comment:
Where does this string come from? can we change this to a constant

Reply:
It's a named key and here used for type narrowing:
https://github.com/firebase/firebase-tools/pull/4931/files/e4ec6a0bfda1cdb917262919a6ed53f274be9858#diff-caf325e78d96d4559be60aff9779ca8a8090299c4078d6366eae20deb9322eb3R52-R55

Since it's a key in the type and used for narrowing, I'd prefer not moving it to a const, even though that does technically work per the type checker. This is just a language quirk but it's essentially no different than writing `response.context`